docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,444 @@ overview page.
+
+Design Choices
+--------------
+
+To choose which solution fits the most our needs, we did a benchmark of some
+of the solutions to compare their performances.
+Our choice for the final design has been greatly motivated by these numbers,
+
+Check `this document`_ for details.
+
+.. _this document: https://docs.google.com/spreadsheets/d/1f2Pz9zBbkDzpHr8KzST3nTcfvWJqgoNcCEd0-4WIeow/edit#gid=0
+
+.. todo::
+
+  export the spreadsheet and include it or do directly a quick recap tab here.
+
+Fluent Bit + Fluentd (optional) + Loki
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Fluent Bit
+""""""""""
+
+We choose Fluent Bit as the collector because it allows to scrape all the logs
+we need (journal & containers), enriches them with the Kubernetes API and it
+has a very low resources footprint.
+
+Moreover, it supports multiple distributors such as Loki, ES, Splunk, etc. at
+the same time, which is a very important point if a user also wants to forward
+the logs to an external log centralization system (e.g. long term archiving).
+
+Check here_ for the official list of supported outputs.
+
+.. _here: https://docs.fluentbit.io/manual/pipeline/outputs
+
+Loki
+""""
+
+Loki has been chosen as the distributor, ingester & querier because, like
+Fluent Bit, it has a really small impact on resources and is very cost
+effective regarding storage needs.
+
+It also uses the LogQL syntax for queries, which is pretty close to what
+we already have with Prometheus and PromQL, so it eases the integration
+in our tools.
+
+Fluentd (optional)
+""""""""""""""""""
+
+Fluentd could be optionally added between Fluent Bit and Loki to offer
+more plugins and reduce the impact of Fluent Bit on workload nodes.
+
+It's not recommended for small platforms (few nodes) as it adds a big overhead
+in resources consumption.
+
+Rejected Design Choices
+-----------------------
+
+Promtail + Loki
+~~~~~~~~~~~~~~~
+
+Promtail has been rejected because, even if it consumes very few resources,
+it can only integrate with Loki and we need something more versatile, with
+the ability to interact with different distributors.
+
+Fluentd + Loki
+~~~~~~~~~~~~~~
+
+This architecture has been rejected because it means we need 1 fluentd instance
+per node, which increases a lot the resources consumption compared to other
+solutions.
+
+Logstash + Elasticsearch
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This architecture is probably the most common one, but it has not been taken
+into consideration because we want to focus on having the minimum resources
+consumption and these components can really hog RAM & CPU. Beats could be used
+as the log collector to reduce the impact of Logstash, but Elasticsearch still
+consumes a lot of resources.
+
+Implementation Details
+----------------------
+
+Deployment
+~~~~~~~~~~
+
+All the components will be deployed using Kubernetes manifests through Salt
+inside a ``metalk8s-logging`` Namespace.
+
+As we already did for other components (e.g. Prometheus Operator), the
```

Comment:
Well, I've considered it, but you can't do a simple architecture with fluentbit + loki (without fluentd).
Otherwise it looks like it's pretty good stuff and would ease many things.
But, if we consider the deployment of fluent as not optional, we could use it.

Reply:
Would need empirical confirmation, but as @xaf-scality suggested, we may be able to configure fluentd as passthrough, reducing its impact on resources.